buildSrc/src/main/java/org/opensearch/gradle/PublishPlugin.java

```diff
@@ -92,7 +92,7 @@ public String call() throws Exception {
                 return String.format(
                     "%s/distributions/%s-%s.pom",
                     project.getBuildDir(),
-                    getArchivesBaseName(project),
+                    pomTask.getName().toLowerCase().contains("zip") ? project.getName() : getArchivesBaseName(project),
```

Comment:
This turned out to be a tricky one: we introduced wrong task for ZIP plugin publication, see https://github.com/opensearch-project/OpenSearch/pull/4696/files#diff-204511a3817dea7312e96afb8aa664eafe003b512a0ac979f4a383b03fbd93c1R134 where the JAR artifact name is being set (but for ZIP artifact, is it `project.getName()`)